pkg/skaffold/instrumentation/meter.go

```diff
@@ -108,30 +141,205 @@ func AddFlag(flag *flag.Flag) {
 }
 
 func ExportMetrics(exitCode int) error {
+	if !shouldExportMetrics {
+		return nil
+	}
 	home, err := homedir.Dir()
 	if err != nil {
 		return fmt.Errorf("retrieving home directory: %w", err)
 	}
 	meter.ExitCode = exitCode
 	meter.Duration = time.Since(meter.StartTime)
-	return exportMetrics(filepath.Join(home, constants.DefaultSkaffoldDir, constants.DefaultMetricFile), meter)
+	return exportMetrics(context.Background(),
+		filepath.Join(home, constants.DefaultSkaffoldDir, constants.DefaultMetricFile),
+		meter)
 }
 
-func exportMetrics(filename string, meter skaffoldMeter) error {
-	if skipExport != "true" || meter.Command == "" {
-		return nil
+func exportMetrics(ctx context.Context, filename string, meter skaffoldMeter) error {
+	logrus.Debug("exporting metrics")
+	p, err := initCloudMonitoringExporterMetrics()
+	if p == nil {
+		return err
 	}
+
 	b, err := ioutil.ReadFile(filename)
+	fileExists := err == nil
 	if err != nil && !os.IsNotExist(err) {
 		return err
 	}
-
 	var meters []skaffoldMeter
 	err = json.Unmarshal(b, &meters)
 	if err != nil {
 		meters = []skaffoldMeter{}
 	}
 	meters = append(meters, meter)
-	b, _ = json.Marshal(meters)
-	return ioutil.WriteFile(filename, b, 0666)
+	if !isOnline {
+		b, _ = json.Marshal(meters)
+		return ioutil.WriteFile(filename, b, 0666)
+	}
+
+	p.Start()
+	for _, m := range meters {
+		createMetrics(ctx, m)
+	}
+	p.Stop()
+
+	err = nil
+	if fileExists {
+		err = os.Remove(filename)
+	}
+	return err
```

Comment:
super nitpicky but I think this could be changed to this
```suggestion
	if fileExists {
		return os.Remove(filename)
	}
	return nil
```